cirq-core/cirq/ops/global_phase_op_test.py

```diff
@@ -271,3 +272,20 @@ def test_gate_op_repr():
 
 def test_gate_global_phase_op_json_dict():
     assert cirq.GlobalPhaseGate(-1j)._json_dict_() == {'coefficient': -1j}
+
+
+def test_parameterization():
+    t = sympy.Symbol('t')
+    gpt = cirq.GlobalPhaseGate(coefficient=t)
+    assert cirq.is_parameterized(gpt)
+    assert cirq.parameter_names(gpt) == {'t'}
+    assert not cirq.has_unitary(gpt)
+    assert gpt.coefficient == t
+    assert (gpt**2).coefficient == t**2
+
+
+@pytest.mark.parametrize('resolve_fn', [cirq.resolve_parameters, cirq.resolve_parameters_once])
+def test_resolve(resolve_fn):
+    t = sympy.Symbol('t')
+    gpt = cirq.GlobalPhaseGate(coefficient=t)
+    assert resolve_fn(gpt, {'t': -1}) == cirq.GlobalPhaseGate(coefficient=-1)
```

Comment:
nit: Also add a test that resolving the parameterized gate with a coefficient whose absolute value is not 1 will raise an error.